silo-vaults/contracts/incentives/claiming-logics/SiloIncentivesControllerCLFactory.sol

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.28;
+
+import {ISiloIncentivesControllerCLFactory} from "../../interfaces/ISiloIncentivesControllerCLFactory.sol";
+import {SiloIncentivesControllerCL} from "./SiloIncentivesControllerCL.sol";
+
+/// @dev Factory for creating SiloIncentivesControllerCL instances
+contract SiloIncentivesControllerCLFactory is ISiloIncentivesControllerCLFactory {
+    mapping(address => bool) public createdInFactory;
+
+    /// @inheritdoc ISiloIncentivesControllerCLFactory
+    function createIncentivesControllerCL(
+        address _vaultIncentivesController,
+        address _siloIncentivesController
+    ) external returns (SiloIncentivesControllerCL logic) {
+        logic = new SiloIncentivesControllerCL(_vaultIncentivesController, _siloIncentivesController);
+
+        createdInFactory[address(logic)] = true;
```

Comment:
what would be the case we need that?

Reply:
I think it's a common pattern for factories.

Reply:
The basic stuff. Just in case we need to check if the contract was created via the factory.